sdcm/ec2_client.py

```diff
@@ -127,22 +129,58 @@ def _request_spot_fleet(self, instance_type, image_id, region_name, network_if,
+            'ap-northeast-2': 'Asia Pacific (Seoul)',
+            'ap-southeast-1': 'Asia Pacific (Singapore)',
+            'ap-southeast-2': 'Asia Pacific (Sydney)',
+            'ap-northeast-1': 'Asia Pacific (Tokyo)',
+            'ca-central-1': 'Canada (Central)',
+            'eu-central-1': 'EU (Frankfurt)',
+            'eu-west-1': 'EU (Ireland)',
+            'eu-west-2': 'EU (London)',
+            'eu-west-3': 'EU (Paris)',
+            'eu-north-1': 'EU (Stockholm)',
+            'sa-east-1': 'South America (Sao Paulo)'
+        }
+
+        pricing = boto3.client('pricing')
+        response = pricing.get_products(
+            ServiceCode='AmazonEC2',
+            Filters=[
+                {'Type': 'TERM_MATCH', 'Field': 'operatingSystem', 'Value': 'Linux'},
+                {'Type': 'TERM_MATCH', 'Field': 'instanceType', 'Value': instance_type},
+                {'Type': 'TERM_MATCH', 'Field': 'preInstalledSw', 'Value': 'NA'},
+                {'Type': 'TERM_MATCH', 'Field': 'tenancy', 'Value': 'Shared'},
+                {'Type': 'TERM_MATCH', 'Field': 'capacitystatus', 'Value': 'Used'},
+                {'Type': 'TERM_MATCH', 'Field': 'location', 'Value': regions_names_map[region_name]}
+            ],
+            MaxResults=10
+        )
+        assert response['PriceList'], "failed to get price for {instance_type} in {region_name}".format(
+            region_name=region_name, instance_type=instance_type)
+        price = response['PriceList'][0]
+        price_dimensions = next(iter(json.loads(price)['terms']['OnDemand'].values()))['priceDimensions']
+        instance_price = next(iter(price_dimensions.values()))['pricePerUnit']['USD']
+        return instance_price
+
+    def _get_spot_price(self, instance_type):
         """
         Calculate spot price for bidding
         :return: spot bid price
         """
-        logger.info('Calculating spot price for bidding')
-        history = self._client.describe_spot_price_history(InstanceTypes=[instance_type], AvailabilityZone=region_name)
-        if 'SpotPriceHistory' not in history or not history['SpotPriceHistory']:
-            raise GetSpotPriceHistoryError("Failed getting spot price history for instance type %s", instance_type)
-        prices = [float(item['SpotPrice']) for item in history['SpotPriceHistory'] if 'Linux/UNIX' in item['ProductDescription']]
-        price_avg = round(sum(prices) / len(prices), 4)
-        price_min = min(prices)
-        price_desired = (price_avg + price_min) / 4
-        if price_desired < price_min:
-            price_desired = price_min
-        price = (dict(min=price_min, max=max(prices), avg=price_avg, desired=round(price_desired, 4)))
+        logger.info('Calculating spot price based on OnDemand price')
+
+        on_demand_price = float(self.get_instance_price(self.region_name, instance_type))
+
+        price = dict(max=on_demand_price, desired=on_demand_price * 0.60)
```

Comment:
`0.60`
Can we define this as a param and put it in our yaml default?
I would like to provide an easy way to override it, in case prices are going up.

Reply:
chanegd, the config is called `spot_max_price`